File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        cur = []
        def dfs(i, total):
            if total > target or i == len(nums):
                return
            if total == target:
                res.append(cur.copy())
                return
            # make decison
            # choose
            cur.append(nums[i])
            dfs(i, total + nums[i])
            # skip
            cur.pop()
            dfs(i+1, total)
        dfs(0, 0)
        return res

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int total = 0;
        int cur = 11000;
        for (int i = 0; i < prices.size(); i++) {
            if (prices[i] > cur) {
                total+= (prices[i] - cur);
            }
            cur = prices[i];
        }
        return total;
    }
};